### PR TITLE
Support theme and image directory

### PR DESCRIPTION
**Pull request by tobynet**
# Motivation 

🐶🚂 We need to support more animals like "longdog" and "locomotive", etc. 

* Add `-t` option for themes. (embedded only)
* Add `-d` option for image directory.
* The default theme is `longcat`, the current 🐈.
* Add a bonus theme `tacgnol`.

## Example

image

## Comming soon... 

The dark-mode is already available. ( https://github.com/tobynet/longcat/commit/bc3cc6216eceb4814063dd46c75b03cb0c506e69 )

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -23,15 +25,67 @@ import (
...
+	t.Tail, _ = loadImage(fs, imgPath("data03.png"))
+	return nil
+}
+
+func (t *Theme) loadThemeFromDir(dir string) error {
+	t.Head, _ = loadImageGlob(dir, "*1.png")
+	t.Body, _ = loadImageGlob(dir, "*2.png")
+	t.Tail, _ = loadImageGlob(dir, "*3.png")
+	return nil
+}
```
Comment (mattn) [inline main.go]: When using statik, loadImage always be succeeded. But loadImageGlob possibly be fail. So should return err.

Comment (tobynet) [reply main.go]: I tried to fix it (tobynet/longcat@b1f04c9cfe2c724cc323ded29145a50eecac68d5)

File: main.go [hunk 2]
```diff
@@ -23,15 +25,76 @@ import (
...
+	Head image.Image
+	Body image.Image
+	Tail image.Image
+}
+
 func loadImage(fs http.FileSystem, n string) (image.Image, error) {
 	f, err := fs.Open(n)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal(fmt.Errorf("theme file does not open: %s", n))
```
Comment (mattn) [inline main.go]: Thanks. please fix this part too. (Yes, I know this part never passed)

```
return nil, fmt.Errorf("theme file does not open: %s", n)
```

File: main.go [hunk 3]
```diff
@@ -23,15 +25,76 @@ import (
...
+}
+
+func (t *Theme) loadTheme(themeName string) error {
+	fs, err := fs.New()
+	if err != nil {
+		return err
+	}
+
+	imgPath := func(s string) string {
+		return filepath.Join("/themes", themeName, s)
```
Comment (mattn) [inline main.go]:
```
return filepath.ToSlash(filepath.Join("/themes", themeName, s))
```

**Comment by mattn** [comment]
Totally LGTM except I pointed.

File: main.go (hunk 2 above)
Comment (tobynet) [reply main.go]: I tried to fix it ( tobynet@8c8f2c4b08e9fb43da00b10038831ef209e6dea3 )

Comment (tobynet) [reply main.go]: I consider necessary for error handlings in `loadImage` too.

e.g. `-t wlongcat` (`/theme/wlongcat/` may not exist in `statik.go`)

Before:

```shell
$ go run main.go -n 2 -H -t wlongcat
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x████████]

goroutine 1 [running]:
main.main()
        ████████████████████████████████████████████longcat/main.go:151 +0x4d5
exit status 2
````

After: (little readable)

```golang
$ go run main.go -n 2 -H -t wlongcat
2019/██/██ ██████ theme file does not open: /themes/wlongcat/data01.png
exit status 1
```

File: main.go (hunk 3 above)
Comment (tobynet) [reply main.go]: Thank you for your review. Using `filepath.ToSlash` is better in this case? Why not ? 

🥴  In `go version go1.10.4 linux/amd64`: 

```golang
gore> filepath.Join("/foo", `../../buzz\mytheme`, "foo.jpg")
"/buzz\\mytheme/foo.jpg"

gore> filepath.ToSlash(filepath.Join("/foo", `\mytheme`, "foo.jpg"))
"/foo/\\mytheme/foo.jpg"

gore> filepath.ToSlash(filepath.Join("/foo", `../../buzz\mytheme`, "foo.jpg"))
"/buzz\\mytheme/foo.jpg"
```

📝 I know that `\`  is allowed in various(?) linux . 

👌 In `go version go1.12.9 windows/amd64`: 

```golang
gore> filepath.Join("/foo", `../../buzz\mytheme`, "foo.jpg")
"\\buzz\\mytheme\\foo.jpg"

gore> filepath.ToSlash(filepath.Join("/foo", `\mytheme`, "foo.jpg"))
"/foo/mytheme/foo.jpg"

gore> filepath.ToSlash(filepath.Join("/foo", `../../buzz\mytheme`, "foo.jpg"))
"/buzz/mytheme/foo.jpg"
```

**Comment by mattn** [comment]
Good to merge for me. Do you have something else to do?

**Comment by mattn** [comment]
Thank you.

**Pull request merged**